src/components/Modal/plugin/ModalPlugin.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <SbModal
+    :is-open-modal="open"
+    :modal-target="modalTarget || '#root'"
```

Comment:
I believe that here you can use the modalTarget value generated by the component itself.